### PR TITLE
Refinement: geodata decoder remove unnecessary GC & export methods for 3rd party

### PR DESCRIPTION
**Pull request by Loyalsoldier**
This PR resolves the performance issue of the new geodata decoder and export methods and errors to 3rd parties.

### CODE REVIEW

**Review by xiaokangwang — APPROVED** [review]
I will merge this code and then make necessary change.
RELEASE BLOCKER

**Review by DuckSoft — COMMENTED** [review]
I see almost no functional change...

**Comment by Loyalsoldier** [comment]
> I see almost no functional change...

The reason why the CPU gets crazy is that too many GCs explicitly.

**Review by DuckSoft — APPROVED** [review]
Looking good.

File: common/geodata/decode.go [hunk 1]
```diff
@@ -23,14 +22,14 @@ import (
...
-	errCodeNotFound                 = errors.New("code not found")
+	ErrFailedToReadBytes            = errors.New("failed to read bytes")
+	ErrFailedToReadExpectedLenBytes = errors.New("failed to read expected length of bytes")
+	ErrInvalidGeodataFile           = errors.New("invalid geodata file")
+	ErrInvalidGeodataVarintLength   = errors.New("invalid geodata varint length")
+	ErrCodeNotFound                 = errors.New("code not found")
 )
 
-func emitBytes(f io.ReadSeeker, code string) ([]byte, error) {
+func EmitBytes(f io.ReadSeeker, code string) ([]byte, error) {
```
Comment (rurirei) [inline common/geodata/decode.go:32]: may not necessary to public the emitBytes, call Decode() for thrid_party is fine.

**Comment by rurirei** [comment]
may not necessary to public the emitBytes. call Decode() for thrid_party is fine.

**Pull request merged**